src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -383,23 +397,32 @@ public void readStdout(int availability)
       try {
          if (availability < 0) {
             outClosed = true;
-            processHandler.onStdout(null);
+            outBuffer.flip();
+            processHandler.onStdout(outBuffer, true);
             return;
          }
          else if (availability == 0) {
```

Comment:
Good catch. We'll now throw a `RuntimeException()` if the buffer has 0 capacity remaining after compacting.